src/extensions/renderer/base/coord-ele-math/coords.js

```diff
@@ -169,13 +169,37 @@ BRp.findNearestElements = function( x, y, interactiveElementsOnly, isTouch ){
       var pts = rs.allpts;
 
       for( var i = 0; i + 3 < pts.length; i += 2 ){
-        if(
-          (math.inLineVicinity( x, y, pts[ i ], pts[ i + 1], pts[ i + 2], pts[ i + 3], width2 ))
-            &&
-          widthSq > ( sqDist = math.sqdistToFiniteLine( x, y, pts[ i ], pts[ i + 1], pts[ i + 2], pts[ i + 3] ) )
-        ){
-          addEle( edge, sqDist );
-          return true;
+        if( rs.edgeType === 'segments' && edge.pstyle( 'taxi-radius' ).value > 0 ) {
```

Comment:
Also:  Hit test code that is curved-taxi-specific should check `edge.pstyle( 'curve-style' ).value === 'taxi'`, just like your drawing code, so that it's explicit and bounded.